README.md

```diff
@@ -1,116 +1,170 @@
+   ```
+
+2. **Forward Pass**: Call the model with input data tensors to receive output predictions.
+
+   ```python
+   input_ids = ...  # Your input tensor here
+   retnet_output = retnet_model(input_ids)
+   transformer_output = transformer_model(input_ids)
+   ```
+
+3. **Parameter Access**: Retrieve model hyperparameters for inspection or further processing.
+
+   ```python
+   retnet_params = retnet_model.get_params()
+   transformer_params = transformer_model.get_params()
+   ```
+
+## Benchmarking and Results
+
+We use EleutherAI's open-source language model evaluation harness to empirically evaluate our models across a suite of different NLP tasks. Run the evaluation suite as follows:
+First, edit the 'tasks' parameter in the YAML file. Specify all tasks you would like to run, e.g.,
+```
+tasks:
+  - "hellaswag"
+  - "winogrande"
 ```
+Alternatively, you can use `tasks: '*'` to run all benchmarks in the suite. Then navigate to the `slurm/run_eval.sh`, copy the script, and substitute your yaml file for the placeholder. Finally, execute:
 
-In `config.yaml`, you can have:
+```
+# Activate environment, if using one
+mamba activate <YOUR_ENV_HERE>
+
+cd /301r_retnet/slurm/
+
+# Give your file a descriptive name, (e.g., 'retnet_40540_run_eval.sh')
+cp run_eval.sh user_slurm/<NAME_OF_NEW_FILE>.sh
 
-```yaml
-learning_rate: 0.001
-batch_size: 64
-num_layers: 4
-embed_dim: 256
+bash <NAME_OF_NEW_FILE>/.sh
 ```
+Results will be sent to a CSV.
+
+## Acknowledgments
+
+We extend our heartfelt gratitude to the following individuals and institutions for their invaluable contributions to this project:
+
+**Nancy Fulda**: Our esteemed instructor, whose guidance and insights have significantly shaped the direction and execution of this research.
+
+**BYU Office of Research Computing**: For providing the computational resources and support that were instrumental in conducting our experiments and analyses.
+
+**Authors of the Original RetNet Paper**: For their pioneering work in the field of deep learning, which inspired our research and provided a solid base for our explorations into RetNet and Transformers.
+
+**Microsoft TorchScale Team**: For developing and maintaining the TorchScale framework, which served as the foundational architecture for our project, enabling us to push the boundaries of what's possible in deep learning research.
 
-### Note
+## Citations
 
-- Replace placeholders like `your_project_model` and `RetNetModel` with actual module and class names from your project.
-- These examples are meant to be adapted to your specific project setup and requirements.
-- The effectiveness of hyperparameter tuning can depend heavily on the specific nature of the dataset and task.
+Our paper is awaiting publication and our full citation will be given soon. Our current citations can be found on the citation section on our paper [Are You Using Retentive Networks?](./301R_Retnet_Paper.pdf).
```

Comment:
Probably take this off. We should probably have either the citations on the readme itself, or something like the following:
> [!NOTE]
> Our paper is awaiting publication and our full citation will be given soon.

Adding on the citations here would be good without a link to the paper itself.

Reply:
While I originally liked the idea of adding our citations into the README.md, the standard for paper repos seems to be add your own citation and let users find the rest in your bibliography (e.g. see anything from [AllenAI](https://github.com/allenai), Google's [BigBench](https://github.com/google/BIG-bench)).

Reply:
I did delete the link to the pdf though.

Reply:
How did you add that " (i) note" block?

Reply:
Remove the # below to see that.
> [!NOTE]
> Our paper is awaiting publication and our full citation will be given soon.

#> [!NOTE]
> Our paper is awaiting publication and our full citation will be given soon.

Reply:
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts